src/overlays/actors/ovl_En_Bigokuta/z_en_bigokuta.c

```diff
@@ -28,99 +43,640 @@ const ActorInit En_Bigokuta_InitVars = {
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2C84.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2CE8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2DAC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2EBC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2F20.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC2F64.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC3054.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC30EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC31EC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC33A4.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC33C0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC3460.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC34A8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC35E8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC3650.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC3930.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC39A0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/EnBigokuta_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC3D48.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/func_80AC4204.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Bigokuta/EnBigokuta_Draw.s")
+void EnBigokuta_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnBigokuta* this = THIS;
+
+    Actor_ProcessInitChain(&this->actor, sInitChain);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gBigOctoSkel, &gBigOctoIdleAnim, this->jointTable,
+                       this->morphTable, 20);
+
+    Collider_InitAndSetCylinder(globalCtx, &this->shellCollider, &this->actor, &sShellCylinderInit);
+    Collider_InitAndSetCylinder(globalCtx, &this->bodyCollider, &this->actor, &sBodyCylinderInit);
+    CollisionCheck_SetInfo(&this->actor.colChkInfo, NULL, &sColChkInfoInit);
+    this->cutscene = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+
+    if (gSaveContext.weekEventReg[20] & 2 ||
+        ((this->actor.params != 0xFF) && Flags_GetSwitch(globalCtx, this->actor.params))) {
+        Actor_MarkForDeath(&this->actor);
+    } else {
+        this->actor.world.pos.y -= 99.0f;
+        EnBigokuta_SetupIdle(this);
+    }
+
+    this->camAt.x = ((Math_SinS(this->actor.home.rot.y) * 66.0f) + this->actor.world.pos.x);
+    this->camAt.y = ((this->actor.home.pos.y - 49.5f) + 42.899998f);
+    this->camAt.z = ((Math_CosS(this->actor.home.rot.y) * 66.0f) + this->actor.world.pos.z);
```

Comment:
```suggestion
    this->camAt.x = Math_SinS(this->actor.home.rot.y) * 66.0f + this->actor.world.pos.x;
    this->camAt.y = (this->actor.home.pos.y - 49.5f) + 42.899998f;
    this->camAt.z = Math_CosS(this->actor.home.rot.y) * 66.0f + this->actor.world.pos.z;
```
I'd also try to see if 42.899998f -> 42.9f matches, but i'm guessing probably not

Reply:
unfortunately not